### PR TITLE
Create build pipelines for Rafter

### PR DESCRIPTION
**Pull request by m00g3n**
**Description**
Create build pipelines for Rafter

Changes proposed in this pull request:

- Makefiles are adjusted to the new repository structure
- Unit tests and formatters are executed and checked

**Related issue(s)**
See also: https://github.com/kyma-project/rafter/issues/3

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -13,63 +24,102 @@ else
...
+	docker tag $(FRONTMATTER_IMG_NAME) $(FRONTMATTER-CI-IMG-NAME)
+	docker push $(FRONTMATTER-CI-IMG-NAME)
+.PHONY: push-frontmatter
 
-# Generate manifests e.g. CRD, RBAC etc.
-manifests: controller-gen
-	$(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+build-asyncapi:
+	docker build -t $(ASYNCAPI_IMG_NAME) -f ${ROOT}/deploy/extension/asyncapi/Dockerfile ${ROOT}
+.PHONY: build-frontmatter
```
Comment (aerfio) [inline Makefile]: compare line 57 and 59

Comment (aerfio) [reply Makefile]: build-asyncapi vs build-frontmatter

File: cmd/uploader/Makefile [hunk 2]
```diff
@@ -0,0 +1,73 @@
+IMG_NAME = rafter-upload-service
```
Comment (aerfio) [inline cmd/uploader/Makefile]: delete this file please

File: Makefile [hunk 3]
```diff
@@ -13,63 +24,102 @@ else
...
+	mkdir -p licenses
+endif
+.PHONY: pull-licenses
 
-# Run go fmt against code
 fmt:
-	go fmt ./...
+	find ${ROOT} -type f -name "*.go" \
+	| egrep -v '_*/automock|_*/testdata|_*export_test.go' \
+	| xargs -L1 go fmt
```
[the remark this answers is not in the document]
Comment (aerfio) [reply Makefile]: discussed offline: no :D

File: Makefile (hunk 1 above)
Comment (m00g3n) [reply Makefile]: it's a `copy-paste` error thx

**Comment by m00g3n** [comment]
> Also I found error in tests on running `docker-build` target
> image

Do you have `kubebuilder` installed?

image

**Comment by m00g3n** [comment]
> Quick review. If you run only build for appropriate image, then build will fail on license copying stage. Maybe better would be separate run for license in dockerfiles, and add for then guard with `true` statement if build fail? What do you thinking about that?

```
RUN go build -a -o main cmd/manager/main.go \
    && mkdir /app \
    && mv ./main /app/main \
    && if [ -f ${BASE_APP_DIR}/licenses ]; then mv ${BASE_APP_DIR}/licenses /app/licenses; fi
```
would work; I think it's a good idea

File: Makefile [hunk 4]
```diff
@@ -13,63 +25,103 @@ else
 GOBIN=$(shell go env GOBIN)
 endif
 
-all: test manager
+all: docker-build
+.PHONY: all
```
Comment (michal-hudy) [inline Makefile]: Write all `.PHONY` targets in one line.

File: Makefile [hunk 5]
```diff
@@ -1,8 +1,20 @@
...
+LICENSES_PATH := ${ROOT}/licenses
 
 # Image URL to use all building/pushing image targets
-APP_NAME ?= rafter-controller-manager
-IMG ?= $(APP_NAME):latest
-IMG-CI = $(DOCKER_PUSH_REPOSITORY)$(DOCKER_PUSH_DIRECTORY)/$(APP_NAME):$(DOCKER_TAG)
+UPLOADER_IMG_NAME := rafter-upload-service
+MANAGER_IMG_NAME := rafter-controller-manager
+FRONTMATTER_IMG_NAME := rafter-frontmatter-service
+ASYNCAPI_IMG_NAME := rafter-asyncapi-service
```
Comment (michal-hudy) [inline Makefile]: It should be possible to override names, so `?=`

File: Makefile [hunk 6]
```diff
@@ -1,8 +1,20 @@
...
+
+IMG-CI-NAME-PREFIX := $(DOCKER_PUSH_REPOSITORY)$(DOCKER_PUSH_DIRECTORY)
```
Comment (michal-hudy) [inline Makefile]: Set default values for `DOCKER_PUSH_REPOSITORY` and `DOCKER_PUSH_DIRECTORY `

File: Makefile [hunk 7]
```diff
@@ -1,8 +1,20 @@
...
+
+UPLOADER-CI-IMG-NAME := $(IMG-CI-NAME-PREFIX)/$(UPLOADER_IMG_NAME):$(DOCKER_TAG)
+MANAGER-CI-IMG-NAME :=  $(IMG-CI-NAME-PREFIX)/$(MANAGER_IMG_NAME):$(DOCKER_TAG)
+FRONTMATTER-CI-IMG-NAME := $(IMG-CI-NAME-PREFIX)/$(FRONTMATTER_IMG_NAME):$(DOCKER_TAG)
+ASYNCAPI-CI-IMG-NAME :=  $(IMG-CI-NAME-PREFIX)/$(ASYNCAPI_IMG_NAME):$(DOCKER_TAG)
```
Comment (michal-hudy) [inline Makefile]: Set default value for `DOCKER_TAG `

File: Makefile (hunk 6 above)
Comment (m00g3n) [reply Makefile]: `IMG-CI-NAME-PREFIX` is used for in `push*` targets invoked only by CI.
The `build*` targets do not use it.

File: Makefile (hunk 7 above)
Comment (m00g3n) [reply Makefile]: same as in `IMG-CI-NAME-PREFIX`

**Pull request merged**